database/import_test.go

```diff
@@ -0,0 +1,324 @@
+				Bits:          402796026,
+				Nonce:         4081063765,
+				State:         "LONGEST_CHAIN",
+				Chainwork:     "2166624730970898396303",
+				CumulatedWork: "255349410425588691745638430",
+				PreviousBlock: "0000000000000000005013e7cc2889ada8b01f24dfc325d1398be82197fc623b",
+			},
+			expectedErrorMessage: "",
+		},
+		{
+			testID: 1,
+			name:   "newer block, should return valid block header",
+			data: testCaseData{
+				blockRecord: [][]string{
+					{"536870912", "e9446d4ebeb301aeb5a2f375ac062bf3581269d783362cf066f08bbe6040a885", "3035389718", "403300437", "1708954423"},
+				},
+				previousBlockHash:  "0000000000000000031817e0b646350cac1b8770d6cba60717e86185cadb15cc",
+				blockHasher:        service.DefaultBlockHasher(),
+				cumulatedChainWork: "409554438998846785912755332",
+				rowIndex:           833233,
+				numberOfBlocks:     1,
+			},
+			expectedBlock: &dto.DbBlockHeader{
+				Height:        833233,
+				Hash:          "00000000000000000676a9b9cdb44820a04c780ca152737124e36341b6c4cdd2",
+				Version:       536870912,
+				MerkleRoot:    "e9446d4ebeb301aeb5a2f375ac062bf3581269d783362cf066f08bbe6040a885",
+				Timestamp:     timestampInLocalTime("2024-02-26 14:33:43+01:00"),
+				Bits:          403300437,
+				Nonce:         3035389718,
+				State:         "LONGEST_CHAIN",
+				Chainwork:     "478151526252246136711",
+				CumulatedWork: "409554917150373038158892043",
+				PreviousBlock: "0000000000000000031817e0b646350cac1b8770d6cba60717e86185cadb15cc",
+			},
+			expectedErrorMessage: "",
+		},
+	}
+
+	for _, tc := range testCases {
+		var result = []dto.DbBlockHeader{}
+		var err error
+		for i := 0; i < tc.data.numberOfBlocks; i++ {
+			block, err := PrepareRecord(tc.data.blockRecord[i], tc.data.previousBlockHash, tc.data.blockHasher, tc.data.cumulatedChainWork, tc.data.rowIndex)
+			if err != nil {
+				t.Errorf("Error while preparing record: %v", err)
+			}
+			result = append(result, *block)
+			tc.data.previousBlockHash = block.Hash
+			tc.data.cumulatedChainWork = block.CumulatedWork
+			tc.data.rowIndex++
+		}
+		assert.Equal[dto.DbBlockHeader](t, result[tc.data.numberOfBlocks-1], *tc.expectedBlock)
+		assert.NoError(t, err)
+	}
+}
+
+func TestPrepareRecordErrorPath(t *testing.T) {
+
+	testCases := []testCase{
+		{
+			testID: 1,
+			name:   "version out of int32 range, should return error",
+			data: testCaseData{
+				blockRecord: [][]string{
+					{"2147483648", "e9446d4ebeb301aeb5a2f375ac062bf3581269d783362cf066f08bbe6040a885", "3035389718", "403300437", "1708954423"},
+				},
+				previousBlockHash:  "0000000000000000031817e0b646350cac1b8770d6cba60717e86185cadb15cc",
+				blockHasher:        service.DefaultBlockHasher(),
+				cumulatedChainWork: "409554438998846785912755332",
+				rowIndex:           833233,
+				numberOfBlocks:     1,
+			},
+			expectedBlock:        nil,
+			expectedErrorMessage: "strconv.ParseInt: parsing \"2147483648\": value out of range",
```

Comment:
It would be good to have more precise message - actually suggesting which value can't be parsed.

The same comment applies to other test cases